src/API/Site/Controllers/MerchantCenter/PhoneNumberController.php

```diff
@@ -0,0 +1,202 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Merchant;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Google\Service\ShoppingContent\Account;
+use Exception;
+use WP_Error;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class PhoneNumberController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class PhoneNumberController extends BaseController {
+
+	/**
+	 * @var Merchant $merchant
+	 */
+	protected $merchant;
+
+	/**
+	 * PhoneNumberController constructor.
+	 *
+	 * @param RESTServer $server
+	 * @param Merchant   $merchant
+	 */
+	public function __construct( RESTServer $server, Merchant $merchant ) {
+		parent::__construct( $server );
+		$this->merchant = $merchant;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'mc/phone-number',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_phone_number_endpoint_read_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->get_phone_number_endpoint_edit_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_update_args(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Get a callback for the phone number endpoint.
+	 *
+	 * @return callable
+	 */
+	protected function get_phone_number_endpoint_read_callback(): callable {
+		return function( $request ) {
+			try {
+				$account = $this->merchant->get_account();
+				return $this->get_phone_number_response( $request, $account );
+			} catch ( Exception $e ) {
+				return new Response( [ 'message' => $e->getMessage() ], $e->getCode() ?: 400 );
+			}
+		};
+	}
+
+	/**
+	 * Get a callback for the edit phone number endpoint.
+	 *
+	 * @return callable
+	 */
+	protected function get_phone_number_endpoint_edit_callback(): callable {
+		return function( Request $request ) {
+			try {
+				$account              = $this->merchant->get_account();
+				$business_information = $account->getBusinessInformation();
+				$business_information->setPhoneNumber( $request['phone_number'] );
+				$account->setBusinessInformation( $business_information );
+				$this->merchant->update_account( $account );
```

Comment:
I think it'd be best to move this logic to a service. The `Merchant` service seems okay but if we add everything there it might get huge since it's too generic. Maybe a new `MerchantVerification` service?

Reply:
Broken out in dae2890c28633ae6f1fb1d124a726c18318b209a